tests/unit/test_metadata.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for metadata module."""
+
+from pathlib import Path
+
+import pytest
+
+from src import exceptions, metadata, types_
+
+from .helpers import assert_substrings_in_string, create_metadata_yaml
+
+
+def test_get_yaml_missing(tmp_path: Path):
+    """
+    arrange: given empty directory
+    act: when get is called with that directory
+    assert: then InputError is raised.
+    """
+    with pytest.raises(exceptions.InputError) as exc_info:
+        metadata.get(path=tmp_path)
+
+    assert metadata.METADATA_FILENAME in str(exc_info.value).lower()
+
+
+@pytest.mark.parametrize(
+    "metadata_yaml_content, expected_error_msg_contents",
+    [
+        pytest.param("", ("empty", metadata.METADATA_FILENAME), id="malformed"),
+        pytest.param(
+            "malformed: yaml:", ("malformed", metadata.METADATA_FILENAME), id="malformed"
+        ),
+        pytest.param("value 1", ("not", "mapping", metadata.METADATA_FILENAME), id="not dict"),
+    ],
+)
+def test_get_yaml_malformed(
+    metadata_yaml_content: str, expected_error_msg_contents: tuple[str, ...], tmp_path: Path
+):
+    """
+    arrange: given directory with metadata.yaml that is malformed
+    act: when get is called with the directory
+    assert: then InputError is raised.
+    """
+    create_metadata_yaml(content=metadata_yaml_content, path=tmp_path)
+
+    with pytest.raises(exceptions.InputError) as exc_info:
+        metadata.get(path=tmp_path)
+
+    assert_substrings_in_string(expected_error_msg_contents, str(exc_info.value).lower())
+
+
+@pytest.mark.parametrize(
+    "metadata_name, metadata_docs_url",
+    [
+        pytest.param("name", "", id="name only"),
```

Comment:
What about the case where the `docs` key is not in `metadata.yaml`? These tests seem to check for the case where it is defined (although empty ion one and a URL in the other case).

This could be achieved by restructuring these parameters to be (1) the contents of the `metadata.yaml` file and (2) the expected `Metadata` potentially making use of the walrus operator to avoid having to define constants multiple times. What do you think? It is kind of similar to the next test

Part of the reason is that currently the tests has a bit of logic in it around `docs=metadata_docs_url or None` which is somewhat implicit, in tests its ok to be a bit more verbose for readability reasons

Reply:
the walrus is awesome! changes applied!